_posts/2017-10-15-What-makes-a-good-README.md

```diff
@@ -0,0 +1,127 @@
+
+What is the project good for? Is it the frontend? Is it the backend? Is it a collection of database migration scripts?
+If it is deployed, here's a good place to put links to the deployments.
+
+> | Name | URL | What is deployed |
+|-------|--------|---------|
+| live | https://your-company.example | Everything that is tagged on master will be auto-deployed here |
+| staging | https://staging.your-company.example | The latest master version, usually volatile and without live data |
+| pull_requests | https://$BRANCH--staging.your-company.example | Every Pull Request is auto-deployed |
+
+
+## 🚀 Prerequisites: What do you need to get started?
+
+> ```shell
+$ brew install nvm gnu-sed watchman  # Mac
+$ yaourt -S nvm                      # Linux
+```
+
+There's always something. You usually need to install a version manager (looking at you, nvm, rvm and such), docker or vagrant. If your dependencies need some specific setup, place it here to make it visible before people start to setup the project.
+
+It's always a good idea to cover the major platforms and ecosystems your colleagues might have.
+
+## ▶️ Running it - You need this all the time
+
+> ```shell
+$ git pull origin   # Get the latest changes
+$ nvm install       # Make sure you're running the node version we need
+$ npm install       # Make sure all dependencies are up-to-date
+$ npm start         # Run the application
+```
+
+Why bother adding `git pull` here you may ask. The answer is, I want everyone in the company to be able to run the latest version of our project on their machine, so I rather mention one more command than leaving them with an old version.
+
+What I also do here (with `nvm install` and `npm install`), is cover updating dependencies should they have changed.
+
+## 🔎 Housekeeping / Troubleshooting
+
+> ```shell
+$ git checkout -f master  # Make sure you're on master and throw away all changes
+$ git clean -xfd          # Undo all untracked changes you did
+$ git pull origin         # Retrieve latest commit
+>
+$ nvm install             # Make sure you're running the node version we need
+$ npm install             # Reinstall all dependencies
+>
+$ npm test                # Run the tests
+$ npm start               # Does this work now?
+```
+
+If someone is running into problems setting up your project, chances are they have unstaged changes, broken dependencies, or happened to pull the latest changes that contain severe bugs. This is the paragraph you need to make sure to narrow down if it's an issue with the codebase or the persons machine.
+
+Make sure to warn them about commands like `git clean -xfd` or `git checkout -f master` though - these will remove all untracked files, even those ignored through `.gitignore`, and undo all changes they did to tracked files - possibly resulting in data loss.
+
```

Comment:
I have good experience with kind of a Q/A format under Troubleshooting. Because that's how it usually works when a new employee runs into troubles with the setup of a project. S/he is asking for help and (should) get appropriate answers. It also gives a more personal touch to a project when you see people interacting. So my tip would be to express issues as questions and solutions as answers. That is also a format which is easy to perpetuate over time or changing team-mates/maintainers.

Reply:
Very good point, I'll add that, thanks!